tests/unit/amo/api/test_index.js

```diff
@@ -106,17 +106,35 @@ describe(__filename, () => {
       mockWindow.verify();
     });
 
-    it('encodes URLs with encodeURI', async () => {
+    it('encodes URL paths with encodeURI', async () => {
       const endpoint = 'project-ă-ă-â-â-日本語';
       mockWindow
         .expects('fetch')
-        .withArgs(sinon.match(encodeURI(`/api/${apiVersion}/${endpoint}/`)))
+        .withArgs(sinon.match(`/api/${apiVersion}/${encodeURI(endpoint)}/`))
         .returns(createApiResponse());
 
       await api.callApi({ endpoint });
       mockWindow.verify();
     });
 
+    // See https://github.com/mozilla/addons-frontend/issues/11447.
+    it('does not double-encode URL querystrings', async () => {
+      const endpoint = 'project-ă-ă-â-â-日本語';
+      const params = { sort: 'users,recommended', category: 'bookmarks' };
+      const expectedQueryString = api.makeQueryString(params);
```

Comment:
While we're here... I don't see tests proving `api.makeQueryString()` encodes values, particularly commas. Maybe we should add one below in the `makeQueryString` tests ? Something like:

```diff
diff --git a/tests/unit/amo/api/test_index.js b/tests/unit/amo/api/test_index.js
index 2b8554deb..0ca96eeb4 100644
--- a/tests/unit/amo/api/test_index.js
+++ b/tests/unit/amo/api/test_index.js
@@ -527,6 +527,11 @@ describe(__filename, () => {
       const query = api.makeQueryString({ some_flag: true });
       expect(query).toEqual('?some_flag=true');
     });
+
+    it('encodes commas in query', () => {
+      const query = api.makeQueryString({ sort: 'foo,bar' });
+      expect(query).toEqual('?sort=foo%2Cbar');
+    });
   });

   describe('fetchAddon', () => {
```

Reply:
Good idea. I've added a test.